prow/test/integration/test/sinker_test.go

```diff
@@ -0,0 +1,178 @@
+// +build e2etest
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	prowjobv1 "k8s.io/test-infra/prow/apis/prowjobs/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestDeletePod(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name       string
+		hasCR      bool
+		wantDelete bool
+	}{
+		{
+			"running-pod",
+			true,
+			false,
+		},
+		{
+			"orphaned-pod",
+			false,
+			true,
+		},
```

Comment:
We'll also want to test some more scenarios like the following:
1. completed, non-orphaned pods are deleted after the `terminatedPodTTL` expires.
1. pods not created by prow are not deleted.
1. prowjobs (not pods) are deleted after `maxProwJobAge` has passed.

I figure this PR is more an initial prototype for integration testing though so we don't need to add these just yet if you'd rather focus on just validating this integration testing pattern.